### DISCUSSION TITLE
How to draw capsules with DebugRenderer

### DISCUSSION THREAD

**Discussion by houseofbits** [jrouwe/JoltPhysics#1342] [Q&A] [answered]
Hello, In my `DebugRenderer` I have implemented `CreateTriangleBatch`  and `DrawGeometry` methods. Triangle meshes are renderer correctly. But it does not render capsule shapes. Am I missing something? 
Also, it crashes when I add `SphereShape` to the scene, here:

`void					DrawGeometry(RMat44Arg inModelMatrix, ColorArg inModelColor, const GeometryRef &inGeometry, ECullMode inCullMode = ECullMode::CullBackFace, ECastShadow inCastShadow = ECastShadow::On, EDrawMode inDrawMode = EDrawMode::Solid) { DrawGeometry(inModelMatrix, inGeometry->mBounds.Transformed(inModelMatrix)`

Looks like inGeometry does not have `mBounds` bounding box.

**Comment by jrouwe** [comment, marked as answer]
It sounds like you're not calling `DebugRenderer::Initialize()` at the end of your DebugRenderer constructor.

Other than that there's very little for me to go on in this ticket.

**Reply by houseofbits** [reply]
Thanks, I have missed exactly this part. Added Initialize to the constructor. But now it fails at reallocation when trying to `CreateQuad`
![Ekrānattēls no 2024-11-22 07-12-44](https://github.com/user-attachments/assets/8c7e547a-446c-4f7b-8d62-2f92301980e1)

![Ekrānattēls no 2024-11-22 07-36-02](https://github.com/user-attachments/assets/aac8f086-cdf3-4518-be96-2ac6fca80458)

Should it be calling reallocate when mElements are nullptr?

**Reply by houseofbits** [reply]
My mistake, I did call constructor before `RegisterDefaultAllocator` therefore allocator was not set.

All good now. Thanks.
